dali/pipeline/data/buffer.h

```diff
@@ -32,7 +32,7 @@ namespace dali {
 class GPUBackend;
 
 // Helper function to get product of dims
-inline Index Product(const vector<Index> &shape) {
+inline Index Volume(const vector<Index> &shape) {
```

Comment:
Idea, can we get a version of this working on iterators, so that we can do things like:
`Volume(shape.begin()+1, shape.end()); // e.g. getting the size of a frame inside a sequence`
Anyway it's probably out of the scope of this PR

Reply:
We can add a overload for this to this function, but yes it is out of the scope of this PR